src/include/cpu/rpp_cpu_filter.hpp

```diff
@@ -0,0 +1,868 @@
+inline void rpp_load_filter_3x3_pln_host(__m256 *pRow, Rpp8u **srcPtrTemp, Rpp32s rowKernelLoopLimit)
+{
+    // irrespective of row location, we need to load 2 rows for 3x3 kernel
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[0], &pRow[0]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[1], &pRow[2]);
+
+    // if rowKernelLoopLimit is 3 load values from 3rd row pointer else set it 0
+    if (rowKernelLoopLimit == 3)
+        rpp_load16_u8_to_f32_avx(srcPtrTemp[2], &pRow[4]);
+    else
+    {
+        pRow[4] = avx_p0;
+        pRow[5] = avx_p0;
+    }
+}
+
+// load function for 5x5 kernel size
+inline void rpp_load_filter_5x5_pln_host(__m256 *pRow, Rpp8u **srcPtrTemp, Rpp32s rowKernelLoopLimit)
+{
+    // irrespective of row location, we need to load 3 rows for 5x5 kernel
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[0], &pRow[0]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[1], &pRow[2]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[2], &pRow[4]);
+
+    for (int k = 3; k < rowKernelLoopLimit; k++)
+        rpp_load16_u8_to_f32_avx(srcPtrTemp[k], &pRow[k * 2]);
+    for (int k = rowKernelLoopLimit; k < 5; k++)
+    {
+        pRow[k * 2] = avx_p0;
+        pRow[k * 2 + 1] = avx_p0;
+    }
+}
+
+// load function for 7x7 kernel size
+inline void rpp_load_filter_7x7_pln_host(__m256 *pRow, Rpp8u **srcPtrTemp, Rpp32s rowKernelLoopLimit)
+{
+    // irrespective of row location, we need to load 4 rows for 7x7 kernel
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[0], &pRow[0]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[1], &pRow[2]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[2], &pRow[4]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[3], &pRow[6]);
+    for (int k = 4; k < rowKernelLoopLimit; k++)
+        rpp_load16_u8_to_f32_avx(srcPtrTemp[k], &pRow[k * 2]);
+    for (int k = rowKernelLoopLimit; k < 7; k++)
+    {
+        pRow[k * 2] = avx_p0;
+        pRow[k * 2 + 1] = avx_p0;
+    }
+}
+
+// load function for 9x9 kernel size
+inline void rpp_load_filter_9x9_pln_host(__m256 *pRow, Rpp8u **srcPtrTemp, Rpp32s rowKernelLoopLimit)
+{
+    // irrespective of row location, we need to load 5 rows for 9x9 kernel
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[0], &pRow[0]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[1], &pRow[2]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[2], &pRow[4]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[3], &pRow[6]);
+    rpp_load16_u8_to_f32_avx(srcPtrTemp[4], &pRow[8]);
+    for (int k = 5; k < rowKernelLoopLimit; k++)
+        rpp_load16_u8_to_f32_avx(srcPtrTemp[k], &pRow[k * 2]);
+    for (int k = rowKernelLoopLimit; k < 9; k++)
+    {
+        pRow[k * 2] = avx_p0;
+        pRow[k * 2 + 1] = avx_p0;
+    }
+}
+
+inline void rpp_load_filter_3x3_pkd_host(__m256 *pRow, Rpp8u **srcPtrTemp, Rpp32s rowKernelLoopLimit)
+{
+    // irrespective of row location, we need to load 2 rows for 3x3 kernel
+    rpp_load8_u8_to_f32_avx(srcPtrTemp[0], &pRow[0]);
+    rpp_load8_u8_to_f32_avx(srcPtrTemp[0] + 8, &pRow[1]);
+    rpp_load8_u8_to_f32_avx(srcPtrTemp[0] + 16, &pRow[2]);
```

Comment:
As discussed before, please try to combine multiple load functions into single load function and see if this gives any performance difference

Reply:
Done